core/src/main/java/cz/xtf/core/openshift/ClusterVersionInfo.java

```diff
@@ -69,14 +70,21 @@ boolean isMajorMinorOnly() {
      * @return true if version is in major.minor.micro format, false if not valid url
      */
     boolean isMajorMinorMicro() {
-        if (openshiftVersion != null) {
-            return versionMatcher.group(2) != null && versionMatcher.group(3) == null;
+        if (openshiftVersion != null && versionMatcher.groupCount() >= 2) {
+            return versionMatcher.group(2) != null;
         }
         return false;
     }
 
+    /**
+     * Check if version is nightly / developer version. It is badly named because anything in
+     * x.y.z-abc is considered nightly.
+     *
+     * @return true, if the version is nightly/developer. Meaning in x.y.z-abc format.
+     * @deprecated There is no replacement, but it can change because of bad name.
+     */
     boolean isNightly() {
```

Comment:
I favor removing (or renaming) this method as it no longer makes sense. Now it can be basically substituted by  `!isMajorMinor && !isMajorMinorMicro`